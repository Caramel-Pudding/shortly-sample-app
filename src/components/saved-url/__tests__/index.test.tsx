import React from "react";

import { render, cleanup } from "@/tests/test-utils";
import { shortcodeAPIResponseStubs } from "@/tests/stubs";

import { SavedUrl } from "..";

describe("SavedUrls Component", () => {
  afterEach(cleanup);

  it("Should render with proper data", () => {
    // * #TEST: ARRANGE
    const { container, queryByText, getByRole } = render(
      <SavedUrl shortUrl={shortcodeAPIResponseStubs[0].result} />,
      {}
    );
    // * #TEST: ASSERT
    expect(
      queryByText(shortcodeAPIResponseStubs[0].result.original_link)
    ).toBeInTheDocument();
    expect(
      queryByText(shortcodeAPIResponseStubs[0].result.short_link)
    ).toBeInTheDocument();
    expect(() => {
      expect(getByRole("button")).toHaveClass("buttonActive");
    }).toThrow();
    expect(container).toMatchSnapshot();
  });

  it("Should detect copied url", () => {
    // * #TEST: ARRANGE
    const { container, getByRole } = render(
      <SavedUrl shortUrl={shortcodeAPIResponseStubs[0].result} />,
      {
        state: {
          urls: {
            urls: [],
            isLoading: false,
            copiedUrl: shortcodeAPIResponseStubs[0].result.short_link,
          },
        },
      }
    );
    // * #TEST: ASSERT
    expect(getByRole("button")).toHaveClass("buttonActive");
    expect(container).toMatchSnapshot();
  });
});
