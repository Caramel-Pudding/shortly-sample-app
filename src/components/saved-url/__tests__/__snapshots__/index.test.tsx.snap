// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedUrls Component Should detect copied url 1`] = `
<div>
  <article
    class="container"
  >
    <span
      class="originalLink"
    >
      https://userinyerface.com/
    </span>
    <section
      class="interactionBlock"
    >
      <span
        class="shortLink"
      >
        shrtco.de/JMwYm
      </span>
      <button
        class="button button normalSizeButton squareButton buttonActive"
        type="button"
      >
        Copied!
      </button>
    </section>
  </article>
</div>
`;

exports[`SavedUrls Component Should render with proper data 1`] = `
<div>
  <article
    class="container"
  >
    <span
      class="originalLink"
    >
      https://userinyerface.com/
    </span>
    <section
      class="interactionBlock"
    >
      <span
        class="shortLink"
      >
        shrtco.de/JMwYm
      </span>
      <button
        class="button button normalSizeButton squareButton"
        type="button"
      >
        Copy
      </button>
    </section>
  </article>
</div>
`;
